src/psyclone/psyir/transformations/intrinsics/matmul2code_trans.py

```diff
@@ -151,7 +151,7 @@ class Matmul2CodeTrans(Operator2CodeTrans):
             do i=1,P
                 R(i,j) = 0.0
                 do ii=1,N
-                    R(i,j) = R(i,j) + A(ii,i) * B(j,ii)
+                    R(i,j) = R(i,j) + A(i,ii) * B(ii,j)
```

Comment:
The dimensions of the arrays in the text before the codeblock doesn't look right (* and it also wasn't before the change I think)

Reply:
Oops, well spotted. I've corrected that now and brought the branch up to master.